cpp/include/raft/neighbors/detail/cagra/graph_core.cuh

```diff
@@ -316,12 +571,468 @@ void sort_knn_graph(raft::resources const& res,
   RAFT_LOG_DEBUG("# Sorting kNN graph time: %.1lf sec\n", time_sort_end - time_sort_start);
 }
 
+template <typename IdxT = uint32_t>
+void mst_optimization(raft::resources const& res,
```

Comment:
Side note- we have an mst implementation in RAFT already but it does not preserve the fixed degree constraint. MST is inherently coupled to neighborhood methods and we are planning to move the existing MST over to cuVS. It would be nice if we could  also expose a public API for this MST implementation, since it's generally useful for cases like this where we want to guarantee (or increase) connectedness. 

We do this trick in our single-linkage clustering algorithm also, with the only caveat that we need to guarantee the smallest edges are added by the MST because the single-linkage has an exact solution.